// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.cmrt.sdk.auto_scaling_service.v1;

import "src/core/common/proto/common.proto";

// For faster allocations of sub-messages.
option cc_enable_arenas = true;

// Handles auto-scaling requests.
service AutoScalingService {
  // If the given instance is in TERMINATING_WAIT state, schedule the
  // termination immediately.
  // If the given instance is not in TERMINATING_WAIT state, do nothing.
  rpc TryFinishInstanceTermination(TryFinishInstanceTerminationRequest) returns (TryFinishInstanceTerminationResponse) {}
}

// Request to try to finish the instance's termination.
message TryFinishInstanceTerminationRequest {
  // Identifier of the instance.
  // In AWS, it is the instance ID.
  // In GCP, it is projects/{project}/zones/{zone}/instances/{instance_id}.
  string instance_resource_id = 1;
  // The name of lifecycle hook.
  string lifecycle_hook_name = 2;
}

// Response of trying to finshe the instance's termination.
message TryFinishInstanceTerminationResponse {
  // The execution result.
  scp.core.common.proto.ExecutionResult result = 1;
  // The termination is scheduled if true.
  bool termination_scheduled = 2;
}
